### ISSUE TITLE
judo.conf

### ISSUE THREAD

**Issue by cthorley** [rollcat/judo#14]
I love judo's approach and aesthetic.  Targeting a specific SSH file is a welcome feature.  It's not uncommon (in an Ansible project, for instance) to have an SSH config that's used for all operations.  Being able to assert a configuration value like this for the project (e.g. `ssh_config = ./ssh/config`) would support a very flexible strategy for capturing host configuration, while keeping the command line invocations very lightweight.  I propose a config file (like `ansible.cnf`) for two reasons:

1. this can be captured in version control, allowing a team to coordinate groups and ssh config relatively transparently
2. a configuration file would permit capturing other command line settings that are consistent for a given project

If I were to implement this, I would probably default the location to `./judo.conf`, and override it with the value of `JUDO_CONFIG` in the environment, if present.  Let me know if you would be open to a PR of this nature, or if this is the incorrect forum for feature requests like this.  Thanks for this tool!

**Comment by rollcat** [comment]
Hi @cthorley! Thanks for the suggestion.

I'm firmly against judo having its own configuration file (or other such parameters), since it would work to break the one aspect of judo that I value very dearly: ease of debugging.

Currently judo closely mirrors the invocation of [`ssh(1)`](https://man.openbsd.org/ssh.1). If `judo -c cmd host` breaks, I can try `ssh -v host cmd` to see what went wrong. This is invaluable e.g. when helping out a coworker who might be less fluent on the "ops" side - most problems I've seen people encounter with judo are, in fact, ssh problems. Ansible also shares this class of problems (on top of a pile of its own), and the piled up hacks in our `ansible.conf` was one of the key motivators to create judo in the first place ;)

I've [added the `-F` option](https://github.com/rollcat/judo/commit/31458eb7617354af3d2b9bfe08a1c69c01dcaaab) because it can be easily passed unchanged to `ssh`; when `judo -F ./ssh_config -c cmd host` breaks, I do `ssh -v -F ./ssh_config host cmd`. It is indeed a bit wordy, but I'm not going to try to hide that I'm merely invoking `ssh` under the hood.

You raise a valid concern though. I do think it would be useful to facilitate sharing `ssh_config` in some way. Perhaps the suggested convention can be to store an `ssh_config` file somewhere convenient in the repository, and advise the use of `Include` (see [`ssh_config(5)`](https://man.openbsd.org/ssh_config.5)) from your personal `~/.ssh/config`?

It's a problem I haven't had yet, since the "work" part of my `~/.ssh/config` boils down to `Host *.company.com`, `User kc` (as it's different from my local login name), and I'm avoiding creating setups that call for anything more complicated.
